src/mlmc/simulation.py

```diff
@@ -16,51 +16,110 @@ def __init__(self, config=None, sim_param=0):
+        self.step = sim_param
         self._input_sample = []
         self._coarse_simulation = None
 
-    def cycle(self):
+    def set_coarse_sim(self, coarse_sim):
+        """
+        Must be called, it is part of initialization.
+        :param coarse_sim:
+        :return:
+        """
         pass
 
-    def n_ops_estimate(self):
+    def simulation_sample(self, tag):
+        # Forward simulation for generated input.
         pass
 
+    def n_ops_estimate(self):
+        # complexity function
+        return self.step
+
     def generate_random_sample(self):
+        # Create new correlated random input for both fine and (related) coarse simulation
         pass
 
-    def get_coarse_sample(self):
-        pass
+    def extract_result(self, sample_dir):
+        """
+        Extract simulation result
+        :param sample_dir: Simulation sample directory
+        :return: simulation result
+        """
+        try:
+            result = self._extract_result(sample_dir)
+            if result is np.nan:
+                raise
+        except:
+            result = np.inf
 
-    def set_previous_fine_sim(self, coarse_sim):
-        pass
+        if result is np.inf:
+            self.mv_failed_realizations(sample_dir)
+
+        return result
 
-    def get_result(self):
+    def _extract_result(self):
         return self._simulation_result
 
     @staticmethod
-    def log_interpolation(sim_param_range, t_level=None):
+    def log_interpolation(sim_param_range, t_level):
         """
         Calculate particular simulation parameter
         :param sim_param_range: Tuple or list of two items, range of simulation parameters
         :param t_level: current level / total number of levels, it means 'precision' of current level fine simulation
         :return: int
         """
-        if t_level is None:
-            return 0
-        else:
-            assert 0 <= t_level <= 1
-            return np.round(sim_param_range[0] ** (1 - t_level) * sim_param_range[1] ** t_level).astype(int)
+        assert 0 <= t_level <= 1
+        return sim_param_range[0] ** (1 - t_level) * sim_param_range[1] ** t_level
 
     @classmethod
-    def make_sim(cls, config, sim_par_range, t_level=None):
+    def factory(cls, step_range, **kwargs):
         """
         Create specific simulation
-        :param config: Simulation configuration
-        :param sim_par_range: Tuple or list of two elements, number of  
-        :param t_level: Simulation parameter of particular simulation
+        :param step_range: Simulation configuration
+        :param **kwargs: Configuration of simulation
         :return: Particular simulation object
         """
-        sim_par = Simulation.log_interpolation(sim_par_range, t_level)
+        return lambda t_level, level_id, kw=kwargs: cls(Simulation.log_interpolation(step_range, t_level), level_id, **kw)
+
+    def mv_failed_realizations(self, sample_dir):
+        """
+        Move failed simulation sample dir
```

Comment:
Make target directory as an optional parameter. Rename to "move_sample_dir" and document main 
use case for moving failed sample dirs.